File: 1.两数之和.ts
/*
 * @lc app=leetcode.cn id=1 lang=typescript
 *
 * [1] 两数之和
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
  const dict: Map<number, number> = new Map();
  const res:number[] = [];
  for (let i = 0; i < nums.length; i++) {
    const v = nums[i];
    const another = target - v;
    if(dict.has(another)){
      const j = dict.get(another) as number;
      res.push(i,j);
      break;
    }
    dict.set(v,i);
  }
  return res;
};
// @lc code=end

File: 1.两数之和.go
package main

/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	var dict map[int]int = make(map[int]int)
	var res []int = []int{}

	for i := 0; i < len(nums); i++ {
		v := nums[i]
		another := target - v
		if _, ok := dict[another]; ok {
			j := dict[another]
			res = []int{i, j}
		}
		dict[v] = i
	}

	return res
}

// @lc code=end
